mongosql/src/algebrizer/definitions.rs

```diff
@@ -605,13 +605,24 @@ impl<'a> Algebrizer<'a> {
                     cache: SchemaCache::new(),
                 })
             }
-            ast::JoinType::Cross | ast::JoinType::Inner => mir::Stage::Join(mir::Join {
-                join_type: mir::JoinType::Inner,
-                left: Box::new(left_src),
-                right: Box::new(right_src),
-                condition,
-                cache: SchemaCache::new(),
-            }),
+            ast::JoinType::Cross | ast::JoinType::Inner => {
+                let join = mir::Stage::Join(mir::Join {
+                    join_type: mir::JoinType::Inner,
+                    left: Box::new(left_src),
+                    right: Box::new(right_src),
+                    condition: None,
+                    cache: SchemaCache::new(),
+                });
+                if let Some(condition) = condition {
```

Comment:
```suggestion
                // The stage_movement optimization will place this condition in the Join if it makes sense. Otherwise,
                // it will move it as early in the pipeline as possible.
                if let Some(condition) = condition {
```
Not sure if I captured this idea exactly correctly, but I think we'd benefit from commenting this to explain why we're doing it.

(This is an amazing idea btw 🔥 )

Reply:
looks good to me! (and thanks)